sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serialization/BasicRelationship.java

```diff
@@ -6,16 +6,20 @@
 import com.azure.core.annotation.Fluent;
 import com.fasterxml.jackson.annotation.JsonAnyGetter;
 import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.HashMap;
 import java.util.Map;
 
+import static com.fasterxml.jackson.annotation.JsonInclude.*;
+
 /**
  * Although relationships have a user-defined schema, these properties should exist on every instance.
  * This is useful to use as a base class to ensure your custom relationships have the necessary properties.
  */
 @Fluent
+@JsonInclude(Include.NON_NULL)
```

Comment:
Service doesn't accept non null values for properties:
`{"$dtId":"dt_Generic_1602401736","$etag":null,"$metadata":{"$model":"dtmi:samples:Building;1"},"AverageTemperature":0,"TemperatureUnit":null}`

```json
{
    "error": {
        "code": "ValidationFailed",
        "message": "Invalid twin specified",
        "details": [
            {
                "code": "ValidationFailed",
                "message": "Expected value of type string for TemperatureUnit on Twin."
            }
        ]
    }
}
```